third_party/docfx-doclet-143274/src/main/java/com/microsoft/build/PackageOverviewFile.java

```diff
@@ -0,0 +1,401 @@
+              .append("</a></td>\n")
+              .append("<td>\n")
+              .append(entry.getValue()[0] != null ? entry.getValue()[0] : "") // Description
+              .append("</td>\n")
+              .append("   </tr>\n");
+        }
+      }
+      classesTableBuilder.append(" </table>\n\n");
+      this.CLASSES_TABLE = classesTableBuilder.toString();
+    }
+    // Check if Interfaces exist in this package
+    boolean containsInterfaces =
+        sortedMap.values().stream().anyMatch(values -> "Interface".equals(values[0]));
+
+    if (containsInterfaces) {
+      this.INTERFACES_TABLE_HEADER = "## Interfaces\n";
+      // Build table containing Interfaces
+      StringBuilder interfacesTableBuilder = new StringBuilder();
+      interfacesTableBuilder
+          .append("<table>\n")
+          .append("   <tr>\n")
+          .append("     <th>\n")
+          .append("Interface")
+          .append("</th>\n")
+          .append("     <th>\n")
+          .append("Description")
+          .append("</th>\n");
+
+      for (Map.Entry<String, String[]> entry : sortedMap.entrySet()) {
+        if (entry.getValue()[1] == "Interface") {
+          interfacesTableBuilder
+              .append("<tr>\n")
+              .append("<td><a href=\"")
+              .append(cloudRADChildElementLinkPrefix + entry.getKey()) // Link to class
+              .append("\">")
+              .append(entry.getKey()) // class name
+              .append("</a></td>\n")
+              .append("<td>\n")
+              .append(entry.getValue()[0] != null ? entry.getValue()[0] : "") // Description
+              .append("</td>\n")
+              .append("   </tr>\n");
+        }
+      }
+      interfacesTableBuilder.append(" </table>\n\n");
+      this.INTERFACES_TABLE = interfacesTableBuilder.toString();
+    }
+
+    // Check if Enums exist in this package
+    boolean containsEnums =
+        sortedMap.values().stream().anyMatch(values -> "Enum".equals(values[1]));
+
+    if (containsEnums) {
+      this.ENUM_TABLE_HEADER = "## Enums\n";
+      // Build table containing Enums
+      StringBuilder enumsTableBuilder = new StringBuilder();
+      enumsTableBuilder
+          .append("<table>\n")
+          .append("   <tr>\n")
+          .append("     <th>\n")
+          .append("Enum")
+          .append("</th>\n")
+          .append("     <th>\n")
+          .append("Description")
+          .append("</th>\n");
+
+      for (Map.Entry<String, String[]> entry : sortedMap.entrySet()) {
+        if (entry.getValue()[1] == "Enum") {
+          enumsTableBuilder
+              .append("<tr>\n")
+              .append("<td><a href=\"")
+              .append(cloudRADChildElementLinkPrefix + entry.getKey()) // Link to class
+              .append("\">")
+              .append(entry.getKey()) // class name
+              .append("</a></td>\n")
+              .append("<td>\n")
+              .append(entry.getValue()[0] != null ? entry.getValue()[0] : "") // Description
```

Comment:
Rather than parsing everything as a Map<String, String[]> (or Map<String, List<String>>) where the values are tracked implicitly, consider making this logic understandable at a glance by creating a quick little POJO:
```java
private class Entry {
  String name;
  String type;
  String description;
}
```

Then all the logic starts looking like:
```java
for (Entry entry : entries) { ...
.append(entry.name) ...

packageItem.getChildrenSummaries().entrySet().stream()
  .sorted(
    Comparator.comparing(entry -> entry.type) // No need to comment about
      .thenComparing(entry -> entry.name)) // what this logic is doing anymore.
  .collect(toList())
```

Reply:
That makes a lot of sense. Refactored!